typed-builder-macro/src/util.rs

```diff
@@ -125,3 +131,88 @@ pub fn expr_to_lit_string(expr: &syn::Expr) -> Result<String, Error> {
         _ => Err(Error::new_spanned(expr, "attribute only allows str values")),
     }
 }
+
+pub enum AttrArg {
+    Flag(Ident),
+    KeyValue(KeyValue),
+    Sub(SubAttr),
+    Not { not: Token![!], name: Ident },
+}
```

Comment:
Feel free to bikeshed, I'm also not sure about having `Not` as a struct variant instead of creating a `AttrNot{...}` struct.

Reply:
I do find value in consistency, but I don't mind struct variants. Even if they were all struct variants it'd be okay. That would require some other changes, which I might do after this PR is merged. So don't change it for now.